src/any_agent/frameworks/any_agent.py

```diff
@@ -63,32 +89,22 @@ def create(
         asyncio.get_event_loop().run_until_complete(agent._load_agent())
         return agent
 
-    @abstractmethod
-    async def _load_agent(self) -> None:
-        """Load the agent instance."""
-
     def run(self, prompt: str) -> Any:
         """Run the agent with the given prompt."""
         return asyncio.get_event_loop().run_until_complete(self.run_async(prompt))
 
+    @abstractmethod
+    async def _load_agent(self) -> None:
+        """Load the agent instance."""
+
     @abstractmethod
     async def run_async(self, prompt: str) -> Any:
         """Run the agent asynchronously with the given prompt."""
 
     @property
     @abstractmethod
-    def tools(self) -> list[Tool]:
```

Comment:
@njbrake I felt like dropping this, we can revisit later if we want to provide some utils to access framework internals.

My reasoning is that:
- It is hard to get a consistent behavior (tools was not accounting for managed_agents, returns different stuff depending on the framework)
- We should focus on standardizing input (AgentConfig) and output #92 , but not "internals".

WDYT?

Reply:
I'm ok with this, just so long as you have some way to assure that the mcp tools are correctly getting merged in with the regular tools, I want to have tests that ensure that both regular and mcp tools are being given to each agent framework.

Reply:
I will update the integration test for that